### ISSUE TITLE
A NULL character can slip through the checks

### ISSUE THREAD

**Issue by perweij** [rpm-software-management/rpmlint#1067]
rpmlint version: 645b9b3 (HEAD at the time of writing)

I noticed that there is a check that can detect a NULL character in certain positions of the spec file, and correctly yield for example `E: forbidden-controlchar-found %changelog:`. But it does not work for all positions - rpmlint will not detect the NULL character (but rpmbuild will).

Expected behaviour: rpmlint should detect the NULL character at all positions that matter.
What actually happens: rpmlint won't detect the NULL character if in the first position (and experimenting further, it also misses it if it's in different positions in the middle of the spec).

Example of a miss: A NULL character first in the spec is not detected:
```
echo -e '\0' > first.spec
cat hello-world.spec  >> first.spec
rpmlint first.spec 

= rpmlint session starts =
rpmlint: 2.4.0
configuration:
    /tmp/rpmlint/venv/lib64/python3.9/site-packages/rpmlint/configdefaults.toml
checks: 27, packages: 1

= 0 packages and 1 specfiles checked; 0 errors, 0 warnings, 0 filtered, 0 badness; has taken 0.0 s =
```

It works when the NULL character is last in the spec:
```
cat hello-world.spec >last.spec
echo -e '\0' >> last.spec
rpmlint last.spec

= rpmlint session starts =
rpmlint: 2.4.0
configuration:
    /tmp/rpmlint/venv/lib64/python3.9/site-packages/rpmlint/configdefaults.toml
checks: 27, packages: 1

last.spec:29: E: forbidden-controlchar-found %changelog: 

= 0 packages and 1 specfiles checked; 1 errors, 0 warnings, 0 filtered, 1 badness; has taken 0.0 s =
```

Here is a standardish hello-world.spec:
```
     1	Name:       hello-world
     2	Version:    1
     3	Release:    1
     4	Buildroot: %{buildroot}
     5	Summary:    Most simple RPM package
     6	License:    FIXME
       
     7	%description
     8	This is my first RPM package, which does nothing.
       
     9	%prep
    10	# we have no source, so nothing here
       
    11	%build
    12	cat > hello-world.sh <<EOF
    13	#!/usr/bin/bash
    14	echo Hello world
    15	EOF
       
    16	%install
    17	mkdir -p %{buildroot}/usr/bin/
    18	install -m 755 hello-world.sh %{buildroot}/usr/bin/hello-world.sh
       
    19	%files
    20	/usr/bin/hello-world.sh
       
    21	%changelog
    22	# let's skip this for now
```
